### PR TITLE
[FSPE-6062] E2E test for Chi Drawer component functionality

### PR DESCRIPTION
**Pull request by joshua22-11**
…eview -- do not merge

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://65be91c3c4d66f04dc8698c9f3d22991ced8056d-chi-pr.enterprise.ctl.io" target="_blank"> https://65be91c3c4d66f04dc8698c9f3d22991ced8056d-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/1330/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://7557e94e3c99702cd5193188d4051da85fa9b881-chi-pr.enterprise.ctl.io" target="_blank"> https://7557e94e3c99702cd5193188d4051da85fa9b881-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/1333/ for more details.

**Review by janibolkvadze — CHANGES_REQUESTED** [review]
The overall structure looks good. There are some false positives related to clicking the "X" button to close the drawer and also some restructuring is necessary. Thank you!

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://629a6eb38e76409be865f1c488a8e3da3437ce23-chi-pr.enterprise.ctl.io" target="_blank"> https://629a6eb38e76409be865f1c488a8e3da3437ce23-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/1360/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://2a1c91ce51fbd4eb7754f83be4a925fa9915deca-chi-pr.enterprise.ctl.io" target="_blank"> https://2a1c91ce51fbd4eb7754f83be4a925fa9915deca-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1378/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://07eb99a890e760f29cb631edb9ef582fbafdc924-chi-pr.enterprise.ctl.io" target="_blank"> https://07eb99a890e760f29cb631edb9ef582fbafdc924-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1384/ for more details.

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+describe('chi-drawer', function() {
+  before(() => {
+    cy.visit('/tests/js/chi-drawer.html');
+  });
+
+  describe('check that click on "X" in left menu drawer, closes drawer', () => {
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:6]: Technically you are testing Open and Close as well. Let's tweak the context of the test group:
```suggestion
  describe('Test open & close behavior of Left positioned Drawer', () => {
```

And inside the describe statement let's add 2 test cases as split scenarios of Open and Close. Something like this:

```
describe('Test open & close behavior of Left positioned Drawer',()=>{
  it('Click on Drawer trigger opens the Drawer', () => {
    /*  */
  });

  it('Click on X button closes the Drawer', () => {
  /*  */
  });
});
```
Basically, split the following test assertions into 2: Open and Close.

File: test/chi/js/chi-drawer.pug [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  .chi-backdrop__wrapper
+    #drawer-1.chi-drawer.-left
+      .chi-drawer__header
+        button.chi-button.-icon.-close(aria-label='Close')
+          .chi-button__content
+            i.chi-icon.icon-x
+      .chi-drawer__content
+        .-px--2.-text Drawer content here
+
+script.
```
Comment (janibolkvadze) [inline test/chi/js/chi-drawer.pug:20]: Let's move all the JS scripts to the bottom of the page inside one consolidated `script.`, please

**Review by janibolkvadze — CHANGES_REQUESTED** [review]
A minor restructuring of test groups is needed and it will be perfect. Thank you!

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://2c6862d608eb37bbba5735bf2f4dc531c45fe2ea-chi-pr.enterprise.ctl.io" target="_blank"> https://2c6862d608eb37bbba5735bf2f4dc531c45fe2ea-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1426/ for more details.

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        .should('have.class', '-left')
+        .should('be.visible');
+    });
+    it('Click on X button closes the Drawer and check for other functionalities', () => {
+      cy.get('#drawer-1')
+        .find('.-close')
+        .click()
+        .parent('.chi-drawer')
+        .should('not.be.visible')
+        .get('#drawer-1')
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:20]: Is this line necessary? I think the current selector is the same element, right?

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+  describe('Test open & close behavior of Left positioned Drawer', () => {
+    it('Click on Drawer trigger opens the Drawer', () => {
+      cy.get('#drawer-trigger-1').click();
+      cy.get('#drawer-1')
+        .should('have.class', '-active')
+        .should('have.class', '-left')
+        .should('be.visible');
+    });
+    it('Click on X button closes the Drawer and check for other functionalities', () => {
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:14]:
```suggestion
    it('Click on X button closes the Drawer', () => {
```

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  before(() => {
+    cy.visit('/tests/js/chi-drawer.html');
+  });
+
+  describe('Test open & close behavior of Left positioned Drawer', () => {
+    it('Click on Drawer trigger opens the Drawer', () => {
+      cy.get('#drawer-trigger-1').click();
+      cy.get('#drawer-1')
+        .should('have.class', '-active')
+        .should('have.class', '-left')
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:11]:
```suggestion
```
This is checking if the class is present in HTML markup.
The proper test would be to check if `left: 0;` CSS is present.
Please use
`.should('have.css' . . .);`

**Review by janibolkvadze — CHANGES_REQUESTED** [review]

**Comment by janibolkvadze** [comment]
@joshua22-11 Probably rebase from master went wrong because I see commits and file changes from merged PRs. Can you try to rebase again to see if this can be cleaned? The current state makes it difficult to review PR related changes

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://d3f639103e72bb6678eccafabf620fdcb517fa6a-chi-pr.enterprise.ctl.io" target="_blank"> https://d3f639103e72bb6678eccafabf620fdcb517fa6a-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1537/ for more details.

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        .should('have.css', 'left','0px')
+        .should('be.visible');
+    });
+    it('Click on X button closes the Drawer and check for other functionalities', () => {
+      cy.get('#drawer-1')
+        .find('.-close')
+        .click()
+        .parent('.chi-drawer')
+        .should('not.be.visible')
+        .get('#drawer-1')
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:20]:
```suggestion
```

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+  });
+
+  describe('Test open & close behavior of Top positioned Drawer', () => {
+    it('Click on Drawer trigger opens the Drawer', () => {
+      cy.get('#drawer-trigger-7').click();
+      cy.get('#drawer-7')
+        .should('have.css', 'top','0px')
+        .should('be.visible');
+    });
+    it('Click on X button closes the Drawer and check for other functionalities', () => {
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:49]:
```suggestion
    it('Click on X button closes the Drawer', () => {
```

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+  });
+
+  describe('Test open & close behavior of Bottom positioned Drawer', () => {
+    it('Click on Drawer trigger opens the Drawer', () => {
+      cy.get('#drawer-trigger-9').click();
+      cy.get('#drawer-9')
+        .should('have.css', 'bottom','0px')
+        .should('be.visible');
+    });
+    it('Click on X button closes the Drawer and check for other functionalities', () => {
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:65]:
```suggestion
    it('Click on X button closes the Drawer', () => {
```

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        .find('.-close')
+        .click()
+        .parent('.chi-drawer')
+        .should('not.be.visible');
+    });
+  });
+
+  describe('Test open & close behavior of Drawer with Backdrop', () => {
+    it('Click on Drawer trigger opens the Drawer', () => {
+      cy.get('#drawer-trigger-11').click();
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:76]:
```suggestion
      cy.get('#drawer-trigger-11')
        .click();
```

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+      cy.get('#drawer-11')
+        .parents('div')
+        .should('have.class', 'chi-backdrop')
+        .get('#drawer-11')
+        .should('have.css', 'bottom','0px')
+        .should('be.visible');
+    });
+    it('Click on X button closes the Drawer and check for other functionalities', () => {
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:84]:
```suggestion
    it('Click on X button closes the Drawer', () => {
```

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+        .find('.-close')
+        .click()
+        .parent('.chi-drawer')
+        .should('not.be.visible');
+    });
+  });
+
+  describe('Test open & close behavior of Drawer with Title', () => {
+    it('Click on Drawer trigger opens the Drawer', () => {
+      cy.get('#drawer-trigger-13').click();
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:95]:
```suggestion
      cy.get('#drawer-trigger-13')
        .click();
```

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+      cy.get('#drawer-13')
+        .find('.chi-drawer__header')
+        .should('be.visible');
+    });
+    it('Click on X button closes the Drawer and check for other functionalities', () => {
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:100]:
```suggestion
    it('Click on X button closes the Drawer', () => {
```

File: cypress/integration/chi-js/chi-drawer.spec.js [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+    it('Click on Drawer trigger opens the Drawer', () => {
+      cy.get('#drawer-trigger-23').click();
+      cy.get('#drawer-23')
+        .parents('div')
+        .should('have.class', 'chi-backdrop')
+        .get('#drawer-23')
+        .should('have.class', '-portal')
+        .should('be.visible');
+    });
+    it('Click on X button closes the Drawer and check for other functionalities', () => {
```
Comment (janibolkvadze) [inline cypress/integration/chi-js/chi-drawer.spec.js:119]:
```suggestion
    it('Click on X button closes the Drawer', () => {
```

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://964f9fddf20b952de2860f30bd2dca4f486c7531-chi-pr.enterprise.ctl.io" target="_blank"> https://964f9fddf20b952de2860f30bd2dca4f486c7531-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1546/ for more details.

**Review by janibolkvadze — APPROVED** [review]

**Review by jllr — APPROVED** [review]

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-l153c5, https://ops.elasticbox.com/services/instances/i-fh2wzt, https://ops.elasticbox.com/services/instances/i-05ybdm, https://ops.elasticbox.com/services/instances/i-ln41yq, https://ops.elasticbox.com/services/instances/i-gyzdhz, https://ops.elasticbox.com/services/instances/i-7z12d5, https://ops.elasticbox.com/services/instances/i-cnpu90, https://ops.elasticbox.com/services/instances/i-6nm2hb

**Pull request merged**